examples/tv-app/android/java/MyUserPrompter-JNI.cpp

```diff
@@ -158,8 +181,37 @@ void JNIMyUserPrompter::PromptForCommissionPasscode(uint16_t vendorId, uint16_t
  */
 void JNIMyUserPrompter::HidePromptsOnCancel(uint16_t vendorId, uint16_t productId, const char * commissioneeName)
 {
-    // TODO
-    ChipLogError(Zcl, "JNIMyUserPrompter::HidePromptsOnCancel Needs Implementation");
+    ChipLogError(Zcl, "JNIMyUserPrompter::HidePromptsOnCancel");
+
+    DeviceLayer::StackUnlock unlock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+    std::string stringCommissioneeName(commissioneeName);
+
+    VerifyOrExit(mJNIMyUserPrompterObject.HasValidObjectRef(), err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mPromptForCommissionOKPermissionMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != nullptr, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        UtfString jniCommissioneeName(env, stringCommissioneeName.data());
```

Comment:
please try not to use utfstring that created extra globalref, you can use env->NewStringUTF(data). I will try to find sometime to cleanup UtfString in later weeks, thanks

Reply:
addressed